### PR TITLE
[codex] Fix file route path encoding

### LINKED ISSUE #112: Bug: file-view routing double-encodes paths
Parent tracker: #91.

The file-view route encodes the path twice, so existing files render as not-found.

## Fix sketch
- Audit the route definition + the navigation call site.
- Encode exactly once (router pushes raw path; the param is then URI-encoded by `vue-router`).
- Add a route test that round-trips a path containing spaces / unicode.

## Acceptance
- [ ] Path encoded once, end-to-end.
- [ ] Route test covers spaces, unicode, and `/` separators.
- [ ] Manual repro confirms existing file opens.

### PR DESCRIPTION
**Pull request by Luis85**
## Summary

Fixes #112.

- Stop pre-encoding file paths before passing them to Vue Router.
- Rename the route param to `filePath` and allow slash-containing paths with a catch-all route segment.
- Stop decoding route params manually in `FileView`, since Vue Router returns decoded params.
- Preserve compatibility for legacy double-encoded file URLs by decoding legacy fully encoded params that still contain percent escapes after Vue Router's first decode.
- Avoid second-decoding current slash-containing paths, preserving literal percent-like filename text such as `%2F`.
- Add route regression tests for spaces, Unicode, slash separators, no `%25` double-encoding, legacy double-encoded URLs, root-level legacy URLs, and literal `%2F` path text.

## Validation

- `npm test -- --run src/ui/router/__tests__/fileRoute.spec.ts`
- `npm run typecheck`
- `npm run lint`
- `npm test`
- `git diff --check`

### CODE REVIEW

File: src/ui/views/FileView.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@ const route = useRoute()
 const router = useRouter()
 const bridge = useBridge()
 
-const filePath = decodeURIComponent(route.params.encodedPath as string)
+const filePath = route.params.filePath as string
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/ui/views/FileView.vue]: Addressed in `bba80cf`. I added `normalizeFileRoutePath()` to preserve legacy double-encoded route compatibility when the decoded route param still contains encoded separators such as `%2F`, and added a regression test for that case.

Verification: `npm test -- --run src/ui/router/__tests__/fileRoute.spec.ts`, `npm run typecheck`, `npm run lint`, `npm test`, and `git diff --check` all pass.

File: src/ui/router/fileRoute.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const LEGACY_ENCODED_SEPARATOR = /%2f/i
+
+export function normalizeFileRoutePath(filePath: string): string {
+  if (!LEGACY_ENCODED_SEPARATOR.test(filePath)) return filePath
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/ui/router/fileRoute.ts]: Addressed in `5734a8d`. `normalizeFileRoutePath()` now decodes legacy percent-encoded params when the route param has no slash separators, which covers root-level legacy URLs like `My%20Note.md` as well as older fully encoded multi-segment params.

Verification: `npm test -- --run src/ui/router/__tests__/fileRoute.spec.ts`, `npm run typecheck`, `npm run lint`, `npm test`, and `git diff --check` all pass.

File: src/ui/router/fileRoute.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+  try {
+    return decodeURIComponent(filePath)
```
[the remark this answers is not in the document]
Comment (Luis85) [reply src/ui/router/fileRoute.ts]: Addressed in `5734a8d`. The compatibility decode now skips route params that already contain `/`, so current paths like `docs/100%2Fready.md` keep the literal `%2F` text instead of being decoded into another separator. Added a regression test for that case.

Verification: `npm test -- --run src/ui/router/__tests__/fileRoute.spec.ts`, `npm run typecheck`, `npm run lint`, `npm test`, and `git diff --check` all pass.
